### PR TITLE
Add X-Spam-Score header to checked mail

### PR DESCRIPTION
**Pull request by yeah**
To enable users to do custom spam filtering based on score, it's helpful to render the actual spam score as a float in a separate header rather than as part of X-Spam-Status where it only appears in a comma separated list.

### CODE REVIEW

**Comment by yeah** [comment]
Here's what the SpamAssassin headers look like without the patch:

```
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on m.launchco.com
X-Spam-Level:
X-Spam-Status: No, score=-2.6 required=5.0 tests=BAYES_00,HTML_MESSAGE,
RCVD_IN_DNSWL_LOW,RCVD_IN_MSPIKE_H2 autolearn=ham autolearn_force=no
version=3.4.0
X-Spam-Report: * -0.7 RCVD_IN_DNSWL_LOW RBL: Sender listed at
http://www.dnswl.org/, low * trust * [a.b.c.d listed in list.dnswl.org]
* -0.0 RCVD_IN_MSPIKE_H2 RBL: Average reputation (+2) * [a.b.c.d listed
in wl.mailspike.net] * 0.0 HTML_MESSAGE BODY: HTML included in message * -1.9
BAYES_00 BODY: Bayes spam probability is 0 to 1% * [score: 0.0000]
```

The score (`-2.6`) is included but it's hard to parse.

With my patch, it will look like this:

```
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on m.launchco.com
X-Spam-Level:
X-Spam-Status: No, score=-2.6 required=5.0 tests=BAYES_00,HTML_MESSAGE,
RCVD_IN_DNSWL_LOW,RCVD_IN_MSPIKE_H2 autolearn=ham autolearn_force=no
version=3.4.0
X-Spam-Report: * -0.7 RCVD_IN_DNSWL_LOW RBL: Sender listed at
http://www.dnswl.org/, low * trust * [a.b.c.d listed in list.dnswl.org]
* -0.0 RCVD_IN_MSPIKE_H2 RBL: Average reputation (+2) * [a.b.c.d listed
in wl.mailspike.net] * 0.0 HTML_MESSAGE BODY: HTML included in message * -1.9
BAYES_00 BODY: Bayes spam probability is 0 to 1% * [score: 0.0000]
X-Spam-Score: -2.6
```

The last line contains the score, easily accessible.

**Comment by JoshData** [comment]
Sounds good, merging.
